jipp-pdl/src/main/kotlin/com/hp/jipp/pdl/pwg/PackBits.kt

```diff
@@ -3,36 +3,43 @@
 
 package com.hp.jipp.pdl.pwg
 
+import com.hp.jipp.pdl.pwg.PwgSettings.Companion.BITS_PER_BYTE
 import java.io.ByteArrayOutputStream
 import java.io.IOException
 import java.io.InputStream
 import java.io.OutputStream
-import java.util.* // ktlint-disable no-wildcard-imports
+import kotlin.math.ceil
 
 /**
  * Encoder/Decoder for the PackBits algorithm described in the Wi-Fi Peer-to-Peer Services Print Technical
  * Specification v1.1
  */
 class PackBits(
-    /** Number of bytes per pixel (1 for grayscale, 3 for RGB) */
-    private val bytesPerPixel: Int,
+    /** Number of bits per pixel (1 for bi-level, 8 for grayscale, 24 for RGB) */
+    private val bitsPerPixel: Int,
     /** Total number of pixels on each horizontal line */
     private val pixelsPerLine: Int
 ) {
 
     /** Reads [inputPixels] until there are no more, writing encoded bytes to [outputBytes] */
     fun encode(inputPixels: InputStream, outputBytes: OutputStream) {
-        EncodeContext(inputPixels, outputBytes, bytesPerPixel, pixelsPerLine).encode()
+        EncodeContext(
+            inputPixels,
+            outputBytes,
+            bitsPerPixel / BITS_PER_BYTE,//bitsPerPixel,
```

Comment:
do we need `ceil(bitsPerPixel.toDouble() / BITS_PER_BYTE).toInt()` here as well?